sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITest.scala

```diff
@@ -460,6 +460,57 @@ class SparkE2EQueryITest
     fieldNames.contains(CosmosTableSchemaInferrer.AttachmentsAttributeName) shouldBe false
   }
 
+  "spark query" can "when forceNullableProperties is false and rows have different schema" in {
```

Comment:
We should also keep/add a test with forceNullableProperties == true. Basically the one where you figured out that we generate a less than ideal query when using a nested property.

Reply:
Not necessarily blocking this PR - can happen seperately

Reply:
Added new test for explain when its true